com.liferay.blade.api/src/com/liferay/blade/api/XMLFile.java

```diff
@@ -21,4 +21,6 @@
 public interface XMLFile extends SourceFile {
 
 	Collection<SearchResult> findElement(String elementName, String elementValue);
+
+	Collection<SearchResult> findLegacyDependency(String groupId, String artifactId);
```

Comment:
We shouldn't add interface here, because not all XMLFiles wil have "legacy dependencies". Methods added here should only be for those that are common to all "XML files"